CONTRIBUTING.md

```diff
@@ -13,14 +13,77 @@ Please fill out either the individual or corporate Contributor License Agreement
 
 Follow either of the two links above to access the appropriate CLA and instructions for how to sign and return it. Once we receive it, we'll be able to accept your pull requests.
 
+## Composition of This Repository and Where/How to Contribute
+
+The repository of the Kubernetes Python client consists of this main repository and a submodule, the [python-base](https://github.com/kubernetes-client/python-base) repository. The main repository contains mostly files that are generated by the OpenAPI generator from [this OpenAPI spec](scripts/swagger.json). The base repo is the utility part of the python client and allows developers to create their own kubernetes clients.
+
+### Where to Submit Your Patch
+
+These folders are automatically generated. You will need to submit a patch to the upstream kubernetes repo [kubernetes](https://github.com/kubernetes/kubernetes) or the OpenAPI generator repo [openapi-generator](https://github.com/OpenAPITools/openapi-generator). This contains:
```

Comment:
Could you clarify "These folders" -- I think "The following folders are automatically generated" would work, and then you can remove the "This contains:" part.

Minor nitpick, but "kubernetes repo" should be uppercased.